server/controllers/jobs-controller.ts

```diff
@@ -71,7 +73,55 @@ export const addJob = async (rule: Rule): Promise<string> => {
 };
 
 export const executeJobs = async (req: Request, res: Response) => {
-    log.info('Executing all available jobs');
+    log.info('job-controller:executeJobs: Executing all available jobs');
     JobRunner.execute();
     res.json({ status: 'started' });
 };
+
+/**
+ * Assigns a Rule to a Job
+ * @param {string} ruleId
+ * @param {string} jobId
+ */
+export const assignRuleToJob = async (ruleId: string, jobId: string) => {
+    try {
+        let job: Job = (await repo.get(jobId)) as Job;
+        job.rules.push(ruleId);
+        await repo.update(jobId, job);
+        log.debug(`Associated rule ${ruleId} with job ${jobId}`);
+        return Promise.resolve();
+    } catch (e) {
+        log.error(e);
+        return Promise.reject(e);
+    }
+};
+
+/**
+ * Disassociate the rule from a job.
+ * Job is deleted if its no longer associated to any of the rules.
+ * @param {string} ruleId
+ */
+export const removeRuleFromJob = async (ruleId: string) => {
+    try {
+        const jobs = await repo.arrayContains('rules', ruleId);
```

Comment:
A bit confusing: you select `rules` but the variable is `jobs`.

Reply:
I dont select rules, I select jobs which have rules [] and look for ruleId in that array